### ISSUE TITLE
[jinja] Single rule violation reported twice, once at parent block level and once at child level (different line numbers)

### ISSUE THREAD

**Issue by bluikko** [ansible/ansible-lint#2757]
##### Summary

If more than one rule violations are detected in a single line and the task is inside a `block` or multiple `block`s then several copies of the same errors are raised, one referring the actual task line number and additional error(s) referring to each of the `block`s that it is inside of.

It seems that ansible-lint starts to look upwards in the file to pick line numbers of previous `block` lines instead of raising just 1 error.

If there are multiple nested `block`s in the tasks file (and multiple occurrences of the same error in that task) then an additional error is created for each of the `block`s, instead of either just reporting a single error affecting the actual task - or reporting multiple errors that refer to the same line number of the actual task:
- For example if there are 4 jinja spacing issues on the same line and the task is inside 3 `block`s then 4 errors are raised: one for the actual task line number and three total for each of the `block`s line numbers.
- If there are 4 jinja spacing issues on the same line and the task is inside 2 `block`s then 3 errors are raised: one for the actual task line number and two total additional copies, one for each of the `block`s line numbers.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.14.0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /ansible/lib64/python3.9/site-packages/ansible
  ansible collection location = /home/.ansible/collections:/usr/share/ansible/collections
  executable location = /ansible/bin/ansible
  python version = 3.9.14 (main, Nov  7 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] (/ansible/bin/python3)
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.9.0 using ansible 2.14.0

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

EL9

##### STEPS TO REPRODUCE

```console (paste below)
---
- name: Message
  debug:
    msg: test msg

- name: Ignore in check mode
  when:
    - not ansible_check_mode | bool
    - not skip_tasks | default('false') | bool
  block:
    - name: Error line numbers should all refer to line eleven
      ansible.builtin.set_fact:
        filelist: "{{ handler_files[0:varx | int] + [namea] + handler_files[vary | int:] }}"
```

##### Desired Behavior

The rule violation report should use the same line number, 11 in this example. Or only 1 error should be raised, regardless of if the task is inside a `block` (or several `block`s).

##### Actual Behavior

ansible-lint starts to look upwards for the previous `block` lines and using line number of a previous such block for the error copy.

In this example task the first error refers line 6 which is incorrect (the previous task) instead of line 11 (the actual task that has the error).

If the task was inside 3 `block`s, and had 5 such "jinja spacing" issues, then 4 errors would be raised:
- One for the actual task (correct).
- 3 errors more with the exact same content but different line number referring the line number of each of the 3 `block`s.

```paste below
jinja[spacing]: Jinja2 spacing could be improved: {{ handler_files[0:varx | int] + [namea] + handler_files[vary | int:] }} -> {{ handler_files[0 : varx | int] + [namea] + handler_files[vary | int :] }}
roles/server/tasks/main.yml:6 Jinja2 template rewrite recommendation: `{{ handler_files[0 : varx | int] + [namea] + handler_files[vary | int :] }}`.

jinja[spacing]: Jinja2 spacing could be improved: {{ handler_files[0:varx | int] + [namea] + handler_files[vary | int:] }} -> {{ handler_files[0 : varx | int] + [namea] + handler_files[vary | int :] }}
roles/server/tasks/main.yml:11 Jinja2 template rewrite recommendation: `{{ handler_files[0 : varx | int] + [namea] + handler_files[vary | int :] }}`.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
AFAIK this is happening only with block-types because the rule is called twice once for the parent task and once for each block inside.

Its current implementation does look in all keys recursively and it should not do that. A PR would be more than welcome here and easy to produce.

@shatakshiiii WDYT about this challenge?
